### DISCUSSION TITLE
Error: too many open files

### DISCUSSION THREAD

**Discussion by dantearmok** [transmission/transmission#5171] [Support]
2.5-GHz Core i5 Mac mini (2012), 16GB RAM, plus an assortment of USB & FW HDs.  Comcast wifi, 100/20 service.
macOS Catalina, Safari, ClamXAV, and PIA (all fully updated).

This is a bit wordy; my apologies.  It's just been one of those evenings!

Over the past day or so, I've had some network outtages, power hits, etc.  This evening, I found that all apps that use the network were stuck (SPOD).  After cleaning that up, and rebooting,...

Launched Transmission (March 5ths nightly), 350+ items all running.  After a minute, it started downloading some items.  But then it started quickly throwing lots of torrents into the Error tab, each saying "Error: too many open files"!  Then it crashed.  This was quite repeatable! 

Crash log and Sample (from subsequent launch) are here:
  https://www.dropbox.com/s/id8j4o4fa296pxs/open%20files%20crash.zip?dl=0

I updated to the March 6 nightly.

Launched Transmission (March 6 nightly).  After a minute, it started downloading some items.  Then it started throwing torrents into the Error tab, each saying too many open files!  It did this fairly slowly, a torrent at a time over a period of about 4 minutes. It didn't crash!  After a few minutes it stopped adding more items to the Error tab and now it seems to be working normally with the rest of the torrents.

Spoke too soon.  I clicked on a group bar, in the main window, then on the Inspector button.  The Inspector window opened but just displayed an empty gray window.  The app crashed about a minute later.

Crash log:
  https://www.dropbox.com/s/0bfazknnu0zfary/Inspector%20is%20blank%20crash.zip?dl=0

If there's more information I can provide, please let me know.

- Dan.

**Comment by dantearmok** [comment]
What are recommended values for Peers> Global maximum connections?  and Peers > Max connections for new transfers?

What exactly means "too many open files"?  Is that a Transmission limit or from macOS?

What is the relationship between these Peers numbers and "too many open files"?

Part of the reason for my asking is that I've found that dramatically increasing the Global Max seems to have eliminated the error message, and the subsequent crash.

Thx,
- Dan.

**Comment by dantearmok** [comment]
I currently have 351 torrents loaded in Transmission.

IF I launch Transmission with all those torrents running, the log gets filled with messages like this:

2023-03-09 18:35:29 +0000 open-files.cc:187 [Error] open-files.cc:187: Couldn't open '/Volumes/Overflow/Torrents Being Downloaded/foldername/filename.mkv': Too many open files (24)

IF I have about 100 of the torrents paused, when I launch Transmission NO errors are generated.  And when I resume the 100, still NO errors are generated.

Running Transmission 4.0.1-dev (8cdb6ec798) -- March 9th's nightly.

hth,
- Dan.

**Comment by ckerr** [comment]
FWIW my testing system tens of thousands of torrents so I'm not seeing the same behavior as @dantearmok wrt hitting some ceiling at 350 torrents :thinking: 

There's an open ticket on this at https://github.com/transmission/transmission/issues/3677 and I've seen the problem myself once, but only saw it after-the-fact and wasn't able to inspect the system to see what files were open.

If anyone has a recipe for repeating this please do follow up in 3677. :bow:

**Comment by dantearmok** [comment]
I'm up to 421 torrents and still seeing the error.

Peers> Global maximum connections
Peers > Max connections for new transfers

What is the relationship between these Peers numbers and "too many open files"?

- Dan.

**Comment by dantearmok** [comment]
I'm down to 109 torrents, and still seeing the error.

My latest -- ran for a bit, hit the "resume all transfers" button, a few secs later the "too many open files" error started.  I tried to quit, but immediately got an SPOD that continued for aro 5 mins.  So I took a sample (below).  It took another 2 to 3 mins for the app to finally quit.

[Sample of Transmission, 5mins SPOD quitting.txt](https://github.com/transmission/transmission/files/11096051/Sample.of.Transmission.5mins.SPOD.quitting.txt)

**Comment by sych0** [comment]
anyone find a solution for this? im running into this routinely

**Comment by dantearmok** [comment]
I haven't seen this issue since April.  @Synch0 can you please provide some details - configuration, etc?

**Comment by rgaufman** [comment]
Any solution to this? - I have 63 transfers (but some of the transfers indeed have many files) and getting this error. Using Transmission 4.0.4 on a Mac Mini M1 running Ventura 13.5.2

**Comment by ckerr** [comment]
Closing as duplicate of #5385
